src/components/Element/Article.jsx

```diff
@@ -0,0 +1,100 @@
+import axios from 'axios';
+import clsx from 'clsx';
+import { useState, useEffect, useRef } from 'react';
+import { Calendar, Tip, Content } from './ArticleElement';
+
+export function Article() {
+  const [article, setArticle] = useState([]);
+  const [sentence, setSentence] = useState({});
+  const [position, setPosition] = useState({});
+  const [link, setLink] = useState('');
+  const markRef = useRef(null);
+
+  useEffect(() => {
+    axios
+      .get(`${process.env.REACT_APP_URL}/api/article`)
+      .then((res) => {
+        setArticle(
+          res.data.articles.reduce((acc, curr) =>
+            curr.content.length > acc.content.length ? curr : acc
+          )
+        );
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!sentence) return;
+    getTipPosition();
+    setLink(translateURL());
+    window.addEventListener('scroll', getTipPosition);
+    window.addEventListener('pointerdown', hideTip);
+
+    return () => {
+      window.removeEventListener('scroll', getTipPosition);
+      window.removeEventListener('pointerdown', hideTip);
+    };
+  }, [sentence]);
+
+  function getTipPosition() {
+    const { scrollY } = window;
+    setPosition({
+      left: markRef.current.offsetLeft,
+      top: markRef.current.offsetTop - scrollY,
+    });
+  }
+
+  function showTip(e) {
+    e.preventDefault();
+    const selection = window.getSelection();
+    const range = selection.getRangeAt(0);
+    const start = range.startOffset;
+    const end = range.endOffset;
+
+    setSentence(() => {
```

Comment:
this is a lot of logic written together, could we refactor whats in setSentence to another function so we have something easy to read. E.g setSentence(sentenceParser())